tcl/cpu/arc/v3_32.tcl

```diff
@@ -0,0 +1,257 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+#  Copyright (C) 2022 Synopsys, Inc.
+#  Shahab Vahedi <shahab@synopsys.com>
+
+source [find cpu/arc/common.tcl]
+
+# Currently 'examine_target' can only read JTAG registers and set properties -
+# but it shouldn't write any of registers - writes will be cached, but cache
+# will be invalidated before flushing after examine_target, and changes will be
+# lost.  Perhaps that would be fixed later - perhaps writes shouldn't be cached
+# after all.  But if write to register is really needed from TCL - then it
+# should be done via "arc jtag" for now.
+proc arc_v3_examine_target { {target ""} } {
+	# Set current target, because OpenOCD event handlers don't do this for us.
+	if { $target != "" } {
+		targets $target
+	}
+
+	# Those registers always exist. DEBUG and DEBUGI are formally optional,
+	# however they come with JTAG interface, and so far there is no way
+	# OpenOCD can communicate with target without JTAG interface.
+	arc set-reg-exists identity pc status32 bta debug eret erbta \
+		erstatus ecr efa
+
+	# 32 core registers + pcl (pc aligned)
+	arc set-reg-exists \
+		r0  r1  r2  r3  r4  r5  r6  r7  r8  r9  r10 r11 r12 \
+		r13 r14 r15 r16 r17 r18 r19 r20 r21 r22 r23 r24 r25 \
+		r26 fp sp ilink gp blink pcl
+
+	# Actionpoints
+	if { [arc get-reg-field ap_build version] == 5 } {
+		set ap_build_type [arc get-reg-field ap_build type]
+		# AP_BUILD.TYPE > 0b0110 is reserved in current ISA.
```

Comment:
Doesn't match ARC HS5x Programmer's Reference Manual (see "25.2.3 Actionpoints Configuration Register, AP_BUILD").

Reply:
You're right, but the only bitfile that I have here still has that version equal to 5, and doesn't work with version 6. Should I change it to 5 here and try to find a newer bitfile where this is fixed?